.github/workflows/checks.yml

```diff
@@ -125,7 +123,7 @@ jobs:
           poetry run poe compile_contracts v1
 
       - name: Upload contracts artifacts
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```

Comment:
note: Bumping because of [soon v3 deprecation](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/). I know we should bump it in separate PR, but then old poetry version would cause problems on CI.